prow/cmd/deck/rerun.go

```diff
@@ -18,21 +18,159 @@ package main
+	branch := refs.BaseRef
+	orgRepo := setRerunOrgRepo(refs, labels)
+	// Get presubmits from Config alone.
+	presubmits := cfg().GetPresubmitsStatic(orgRepo)
+	for _, job := range presubmits {
+		job := job
+		if !job.CouldRun(branch) { // filter out jobs that are not branch matching
+			continue
+		}
+		if job.Name == name {
+			if presubmitJob != nil {
+				return nil, nil, nil, fmt.Errorf("%s matches multiple prow jobs from orgRepo %q", name, orgRepo)
+			}
+			presubmitJob = &job
+		}
+	}
+	if presubmitJob == nil {
+		return nil, nil, nil, fmt.Errorf("failed to find associated presubmit job %q from orgRepo %q", name, orgRepo)
+	}
+
+	prowJobSpec := pjutil.PresubmitSpec(*presubmitJob, *refs)
+	return &prowJobSpec, presubmitJob.Labels, presubmitJob.Annotations, nil
+}
+
+func getPostsubmitSpec(cfg config.Getter, name string, refs *prowapi.Refs, labels map[string]string) (*v1.ProwJobSpec, map[string]string, map[string]string, error) {
+	// postsubmit jobs require Refs to be set
+	if err := verifyRerunRefs(refs); err != nil {
+		return nil, nil, nil, err
+	}
+	var postsubmitJob *config.Postsubmit
+	branch := refs.BaseRef
+	orgRepo := setRerunOrgRepo(refs, labels)
+	postsubmits := cfg().GetPostsubmitsStatic(orgRepo)
+	for _, job := range postsubmits {
+		job := job
+		if !job.CouldRun(branch) { // filter out jobs that are not branch matching
+			continue
+		}
+		if job.Name == name {
+			if postsubmitJob != nil {
+				return nil, nil, nil, fmt.Errorf("%s matches multiple prow jobs from orgRepo %q", name, orgRepo)
+			}
+			postsubmitJob = &job
+		}
+	}
+	if postsubmitJob == nil {
+		return nil, nil, nil, fmt.Errorf("failed to find associated postsubmit job %q from orgRepo %q", name, orgRepo)
+	}
+	prowJobSpec := pjutil.PostsubmitSpec(*postsubmitJob, *refs)
+	return &prowJobSpec, postsubmitJob.Labels, postsubmitJob.Annotations, nil
+}
+
+func getPeriodicSpec(cfg config.Getter, name string) (*v1.ProwJobSpec, map[string]string, map[string]string, error) {
+	var periodicJob *config.Periodic
+	for _, job := range cfg().AllPeriodics() {
+		if job.Name == name {
+			// Directly followed by break, so this is ok
+			// nolint: exportloopref
+			periodicJob = &job
+			break
+		}
+	}
+	if periodicJob == nil {
+		return nil, nil, nil, fmt.Errorf("failed to find associated periodic job %q", name)
+	}
+	prowJobSpec := pjutil.PeriodicSpec(*periodicJob)
+	return &prowJobSpec, periodicJob.Labels, periodicJob.Annotations, nil
+}
+
+func getProwJobSpec(pjType v1.ProwJobType, cfg config.Getter, name string, refs *prowapi.Refs, labels map[string]string) (*v1.ProwJobSpec, map[string]string, map[string]string, error) {
+	switch pjType {
+	case prowapi.PeriodicJob:
+		return getPeriodicSpec(cfg, name)
+	case prowapi.PresubmitJob:
+		return getPresubmitSpec(cfg, name, refs, labels)
+	case prowapi.PostsubmitJob:
+		return getPostsubmitSpec(cfg, name, refs, labels)
+	default:
+		return nil, nil, nil, errors.New("Could not create new prowjob: Invalid prowjob type")
```

Comment:
Nit: Add the invalid job type to the error msg

Reply:
Updated!